scrapy/signalmanager.py

```diff
@@ -3,15 +3,36 @@
 
 from scrapy.utils import signal as _signal
 from scrapy.exceptions import ScrapyDeprecationWarning
-from scrapy.dispatch.robustapply import robustApply
+from scrapy.dispatch.robustapply import robust_apply as _robust_apply
 from scrapy.dispatch.utils import func_accepts_kwargs
-
+from scrapy.dispatch import Signal as _Signal
 
 class SignalManager(object):
 
     def __init__(self, sender=None):
         self.sender = sender
-        self.signals = {}
+        self._patched_receivers = {}
+        self._signal_proxies = {}
+
+    def _ensure_signal(self, signal):
+        """
+        This method is for backward compatability for any custom signals
+        that might be in use in third party extensions which is still using the
+        standard python objects as signals. This method returns a Signal()
+        class instance and saves it to proxy the user defined signal.
+        """
+        if isinstance(signal, _Signal):
+            return signal
+        # Ensure we got at least an old style signal
+        warnings.warn("Signals in scrapy are no longer instances of the basic "
+                      "python object but rather the Signal class defined in "
+                      "the scrapy.dispatch module. Please refer to the Signal "
+                      "API documentation.",
```

Comment:
It would be good if the warning were formulated as instruction instead of description, e.g. `Custom signals should be instances of scrapy.dispatch.Signal.`